products/firebase/api.yaml

```diff
@@ -0,0 +1,72 @@
+# Copyright 2018 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Firebase
+display_name: Firebase
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://firebase.googleapis.com/v1/
+  - !ruby/object:Api::Product::Version
+    name: beta
+    base_url: https://firebase.googleapis.com/v1beta1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+async: !ruby/object:Api::OpAsync
+  operation: !ruby/object:Api::OpAsync::Operation
+    path: 'name'
+    base_url: '{{op_id}}'
+    wait_ms: 1000
+  result: !ruby/object:Api::OpAsync::Result
+    path: 'response'
+    resource_inside_response: true
+  status: !ruby/object:Api::OpAsync::Status
+    path: 'done'
+    complete: true
+    allowed:
+      - true
+      - false
+  error: !ruby/object:Api::OpAsync::Error
+    path: 'error'
+    message: 'message'
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Project'
+    min_version: beta
+    base_url: projects/{{project}}
+    create_url: projects/{{project}}:addFirebase
+    input: true
+    description: |
+      A Google Cloud Firebase instance. This enables Firebase resources on a given google project.
+      Since a FirebaseProject is actually also a GCP Project, a FirebaseProject uses underlying GCP
+      identifiers (most importantly, the projectId) as its own for easy interop with GCP APIs.
+
+      Once Firebase has been added to a Google Project it cannot be removed.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation':
+          'https://firebase.google.com/'
+      api: 'https://firebase.google.com/docs/projects/api/reference/rest/v1beta1/projects'
+    properties:
```

Comment:
There are a couple other properties that come back in the response, but they all appear to have the same value and all of them are the project-id. So I wasn't sure if it was worth adding them.

ex:
```
{
  "projectId": "tf-testjeyulibspx",
  "projectNumber": "684476938629",
  "displayName": "tf-testjeyulibspx",
  "name": "projects/tf-testjeyulibspx",
  "resources": {
    "hostingSite": "tf-testjeyulibspx",
    "realtimeDatabaseInstance": "tf-testjeyulibspx"
  }
}
```